src/DataSource/DB.hs

```diff
@@ -0,0 +1,162 @@
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+-- The idea behind this module and doing orphan instances is that
+-- if we ever want to switch to another implementation, we should
+-- just remove this module, not rewrite all the types and change instances.
+-- This way, we have real separation.
+
+module DataSource.DB
+    ( withDatabase
+    , withProdDatabase
+    , cachedZendeskLayer
+    ) where
+
+import           Universum hiding (All)
+
+import           Data.Text (split)
+
+import           Database.SQLite.Simple
+import           Database.SQLite.Simple.FromField
+import           Database.SQLite.Simple.Internal
+import           Database.SQLite.Simple.Ok
+import           Database.SQLite.Simple.ToField
+
+import           DataSource.Types (Attachment (..), AttachmentContent (..), AttachmentId (..),
+                                   Comment (..), CommentBody (..), CommentId (..), Config,
+                                   TicketId (..), TicketInfo (..), TicketStatus (..),
+                                   TicketTags (..), TicketURL (..), UserId (..), ZendeskLayer (..))
+
+
+-- https://ocharles.org.uk/blog/posts/2014-08-07-postgresql-simple-generic-sop.html
+
+-- | The cached Zendesk layer. We are using a simple SQLite DB behind the scenes
+-- that we need to sync occasionaly.
+cachedZendeskLayer :: (MonadIO m, MonadReader Config m) => ZendeskLayer m
+cachedZendeskLayer = ZendeskLayer
+    { zlGetTicketInfo           = liftIO . getTicketInfoByTicketId
+    , zlListAssignedTickets     = liftIO . getAllAssignedTicketsByUser
+    , zlListRequestedTickets    = liftIO . getAllRequestedTicketsByUser
+    , zlGetAttachment           = liftIO . DataSource.DB.getAttachmentContent
+    , zlGetTicketComments       = liftIO . getTicketComments
+    , zlPostTicketComment       = error "We won't use this for now!"
+    }
+
+-- Database instances
+-- FROM
+
+instance FromField TicketId where
+    fromField (Field (SQLInteger tId) _)    = Ok . TicketId . fromIntegral $ tId
+    fromField f                             = returnError ConversionFailed f "need a integer, ticket id"
```

Comment:
Grammar error. "need `an` integer, ticket id" .  Other error message has same issues as well

Reply:
Interesting, I remember I read somewhere they can be interchanged, but it doesn't seem to be so - https://www.quickanddirtytips.com/education/grammar/a-versus-an